### PR TITLE
OIDC updates

### PR DESCRIPTION
**Pull request by rosieyohannan**
# Description
Update OIDC doc

# Reasons
There is a new OIDC token available, and new additional claims

# Content Checklist
Please follow our style when contributing to CircleCI docs. Our style guide is here: [https://circleci.com/docs/style/style-guide-overview](https://circleci.com/docs/style/style-guide-overview).

Please take a moment to check through the following items when submitting your PR (this is just a guide so will not be relevant for all PRs) 😸:

- [ ] Break up walls of text by adding paragraph breaks.
- [ ] Consider if the content could benefit from more structure, such as lists or tables, to make it easier to consume.
- [ ] Keep the title between 20 and 70 characters.
- [ ] Consider whether the content would benefit from more subsections (h2-h6 headings) to make it easier to consume.
- [ ] Check all headings h1-h6 are in sentence case (only first letter is capitalized).
- [ ] Is there a "Next steps" section at the end of the page giving the reader a clear path to what to read next?
- [ ] Include relevant backlinks to other CircleCI docs/pages.

### CODE REVIEW

File: jekyll/_cci2/openid-connect-tokens.adoc [hunk 1]
```diff
@@ -54,7 +54,11 @@ The OpenID Connect ID token contains the following standard https://openid.net/s
...
+a| The subject. This identifies who is running the CircleCI job and where. `$CIRCLE_OIDC_TOKEN_2` also includes information about the source of change.
+
+For `$CIRCLE_OIDC_TOKEN` its value is: `"org/ORGANIZATION_ID/project/PROJECT_ID/user/USER_ID"`, a string, where `ORGANIZATION_ID`, `PROJECT_ID`, and `USER_ID` are UUIDs that identify the CircleCI organization, project, and user, respectively. The user is the CircleCI user that caused this job to run. 
+
+For `$CIRCLE_OIDC_TOKEN_2` its value is: `"org/ORGANIZATION_ID/project/PROJECT_ID/user/USER_ID/vcs-origin/VCS_ORIGIN/vcs-ref/VCS_REF"`, a string, where `ORGANIZATION_ID`, `PROJECT_ID`, and `USER_ID` are UUIDs that identify the CircleCI organization, project, and user, respectively. The user is the CircleCI user that caused this job to run. `VCS_ORIGIN` and `VCS_REF` are strings that identify the repo URL and reference to the change that caused the job to run.
```
Comment (rosieyohannan) [inline jekyll/_cci2/openid-connect-tokens.adoc:61]: I tried to clarify what the VCS elements do, this needs checking @nathanfishcircleci

**Comment by rosieyohannan** [comment]
@nathanfishcircleci I've used the info I was missing from your PR. The main difference is that I have just kept one table for both tokens and highlighted the differenced just for `sub`. Let me know if this is ok. If you think there are going to be many more tokens added over time it might be better for scalability if I create separate tables as you have.

**Comment by nathanfishcircleci** [comment]
@rosieyohannan that looks good to me.

File: jekyll/_cci2/openid-connect-tokens.adoc [hunk 2]
```diff
@@ -78,6 +82,12 @@ The OpenID Connect ID token also contains some https://openid.net/specs/openid-c
 | `oidc.circleci.com/project-id`
 | The ID of the project in which the job is running. Its value is a string containing a UUID identifying the CircleCI project.
 
+| `oidc.circleci.com/vcs-origin`
+| The URL of the repo that triggered the pipeline. Its value is a string similar to `github.com/orgnanization-123/repo-1`.
```
Comment (kbatuigas) [inline jekyll/_cci2/openid-connect-tokens.adoc:86]:
```suggestion
| The URL of the repo that triggered the pipeline. Its value is a string similar to `github.com/organization-123/repo-1`.
```

File: jekyll/_cci2/openid-connect-tokens.adoc [hunk 3]
```diff
@@ -32,16 +32,16 @@ workflows:
...
 
-The https://openid.net/specs/openid-connect-core-1_0.html#Terminology[OpenID Provider] is unique to your organization. The URL is `\https://oidc.circleci.com/org/ORGANIZATION_ID`, where `ORGANIZATION_ID` is the organization ID (a universally unique identifier) that represents your organization. You can find your CircleCI organization ID by navigating to **Organization Settings > Overview** on the https://app.circleci.com/[CircleCI web app].
+The https://openid.net/specs/openid-connect-core-1_0.html#Terminology[OpenID Provider] is unique to your organization. The URL is `\https://oidc.circleci.com/org/ORGANIZATION_ID`, where `ORGANIZATION_ID` is the organization ID (a universally unique identifier) that represents your organization. You can find your CircleCI organization ID by navigating to **Organization Settings > Overview** in the https://app.circleci.com/[CircleCI web app].
```
Comment (kbatuigas) [inline jekyll/_cci2/openid-connect-tokens.adoc:37]: > The URL is `\https://oidc.circleci.com/org/ORGANIZATION_ID`, where `ORGANIZATION_ID` is the organization ID (a universally unique identifier) that represents your organization.

Wanted to double check that we mean to have `\` in the URL

**Review by kbatuigas — CHANGES_REQUESTED** [review]
Just a couple of minor comments

File: jekyll/_cci2/openid-connect-tokens.adoc [hunk 4]
```diff
@@ -11,12 +11,12 @@ contentTags:
...
 :toc-title:
 
-In jobs using a <<contexts#,context>>, CircleCI provides an OpenID Connect ID (OIDC) token in an environment variable. A job can use this to access compatible cloud services without a long-lived credential stored in CircleCI.
+In jobs using a xref:contexts#[context], CircleCI provides OpenID Connect ID (OIDC) tokens in environment variables. A job can use these tokens to access compatible cloud services without long-lived credentials being stored in CircleCI.
 
 [#openid-connect-id-token-availability]
 == OpenID Connect ID token availability
 
-In CircleCI jobs that use at least one context, the OpenID Connect ID token is available in the environment variable `$CIRCLE_OIDC_TOKEN`.
+In CircleCI jobs that use at least one context, the OpenID Connect ID tokens are available in the environment variables: `$CIRCLE_OIDC_TOKEN` and `$CIRCLE_OIDC_TOKEN_2`. The `$CIRCLE_OIDC_TOKEN_2` token includes a different format for the `sub` claim, which is highlighted in the table below.
```
Comment (michael-webster) [inline jekyll/_cci2/openid-connect-tokens.adoc:19]: The new token name should be `CIRCLE_OIDC_TOKEN_V2`
```suggestion
In CircleCI jobs that use at least one context, the OpenID Connect ID tokens are available in the environment variables: `$CIRCLE_OIDC_TOKEN` and `$CIRCLE_OIDC_TOKEN_V2`. The `$CIRCLE_OIDC_TOKEN_V2` token includes a different format for the `sub` claim, which is highlighted in the table below.
```

File: jekyll/_includes/snippets/built-in-env-vars.adoc [hunk 5]
```diff
@@ -41,6 +41,10 @@
 | String
 | An OpenID Connect token signed by CircleCI which includes details about the current job. Available in jobs that use a context.
 
+| `CIRCLE_OIDC_TOKEN_2`
```
Comment (michael-webster) [inline jekyll/_includes/snippets/built-in-env-vars.adoc:44]:
```suggestion
| `CIRCLE_OIDC_TOKEN_V2`
```

File: jekyll/_cci2/openid-connect-tokens.adoc (hunk 1 above)
Comment (michael-webster) [inline jekyll/_cci2/openid-connect-tokens.adoc:61]:
```suggestion
For `$CIRCLE_OIDC_TOKEN_V2` its value is: `"org/ORGANIZATION_ID/project/PROJECT_ID/user/USER_ID/vcs-origin/VCS_ORIGIN/vcs-ref/VCS_REF"`, a string, where `ORGANIZATION_ID`, `PROJECT_ID`, and `USER_ID` are UUIDs that identify the CircleCI organization, project, and user, respectively. The user is the CircleCI user that caused this job to run. `VCS_ORIGIN` and `VCS_REF` are strings that identify the repo URL and reference to the change that caused the job to run.
```

**Comment by michael-webster** [comment]
@rosieyohannan I am not sure how you normally include examples, but I put together an example of a trust policy that limits access to specific branches for any org in a project:

Example AWS Trust policy that will restrict the `AssumeRoleWithWebIdentity` to pipelines from any project in an organization and github org that runs on the main branch.

Here we limit to a specific organization id and a github org name. You can further restrict access to specific project orgs, or tags, by chaning the string format

```json
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Effect": "Allow",
            "Principal": {
                "Federated": "arn:aws:iam::269508603282:oidc-provider/oidc.circleci.com/org/6b376885-8f20-45b2-ba2f-c27c0919bc6d"
            },
            "Action": "sts:AssumeRoleWithWebIdentity",
            "Condition": {
                "StringLike": {
                    "oidc.circleci.com/org/6b376885-8f20-45b2-ba2f-c27c0919bc6d:sub": "org/6b376885-8f20-45b2-ba2f-c27c0919bc6d/project/*/user/*/vcs-origin/github.com/my-org/*/vcs-ref/refs/heads/main"
                }
            }
        }
    ]
}```

**Review by michael-webster — COMMENTED** [review]
A few suggestions to change the name of the token. I also left a general comment with an example AWS trust policy that does a branch restriction for an org.

File: jekyll/_cci2/openid-connect-tokens.adoc (hunk 3 above)
Comment (rosieyohannan) [reply jekyll/_cci2/openid-connect-tokens.adoc:37]: strange, yeah will remove!

**Comment by rosieyohannan** [comment]
> @rosieyohannan I am not sure how you normally include examples, but I put together an example of a trust policy that limits access to specific branches for any org in a project:
> 
> Example AWS Trust policy that will restrict the `AssumeRoleWithWebIdentity` to pipelines from any project in the `6b376885-8f20-45b2-ba2f-c27c0919bc6d` CircleCI organization and the github org named `my-org`, that runs on the main branch.
> 
> Here we limit to a specific organization id and a github org name. You can further restrict access to specific project orgs, or tags, by changing the string format
> 
> ```json
> {
>     "Version": "2012-10-17",
>     "Statement": [
>         {
>             "Effect": "Allow",
>             "Principal": {
>                 "Federated": "arn:aws:iam::269508603282:oidc-provider/oidc.circleci.com/org/6b376885-8f20-45b2-ba2f-c27c0919bc6d"
>             },
>             "Action": "sts:AssumeRoleWithWebIdentity",
>             "Condition": {
>                 "StringLike": {
>                     "oidc.circleci.com/org/6b376885-8f20-45b2-ba2f-c27c0919bc6d:sub": "org/6b376885-8f20-45b2-ba2f-c27c0919bc6d/project/*/user/*/vcs-origin/github.com/my-org/*/vcs-ref/refs/heads/main"
>                 }
>             }
>         }
>     ]
> }
> ```

Thank you @michael-webster! I will make a separate ticket so we can think about whether this requires changing the how-to guide, or if we just need to incorporate an example

File: jekyll/_cci2/openid-connect-tokens.adoc [hunk 6]
```diff
@@ -54,7 +54,11 @@ The OpenID Connect ID token contains the following standard https://openid.net/s
 | The issuer. The issuer is specific to the CircleCI organization in which the job is being run. Its value is `"https://oidc.circleci.com/org/ORGANIZATION_ID"`, a string, where `ORGANIZATION_ID` is a UUID identifying the current job's project's organization.
 
 | `sub`
-| The subject. This identifies who is running the CircleCI job and where. Its value is `"org/ORGANIZATION_ID/project/PROJECT_ID/user/USER_ID"`, a string, where `ORGANIZATION_ID`, `PROJECT_ID`, and `USER_ID` are UUIDs that identify the CircleCI organization, project, and user, respectively. The user is the CircleCI user that caused this job to run.
+a| The subject. This identifies who is running the CircleCI job and where. `$CIRCLE_OIDC_TOKEN_2` also includes information about the source of change.
```
Comment (rosieyohannan) [inline jekyll/_cci2/openid-connect-tokens.adoc:57]:
```suggestion
a| The subject. This identifies who is running the CircleCI job and where. `$CIRCLE_OIDC_TOKEN_V2` also includes information about the source of change.
```

**Review by kbatuigas — APPROVED** [review]
LGTM!

**Pull request merged**